app/src/main/java/com/ae/apps/tripmeter/fragments/TripsListFragment.java

```diff
@@ -103,10 +154,30 @@ public void onClick(View v) {
                 showAddTripDialog();
             }
         });
-
         return view;
     }
 
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSIONS_REQUEST_READ_CONTACTS: {
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    //TODO: Reload this Fragment.Otherwise the FAB won't work
```

Comment:
The actual issue is caused due to reading of contacts during instantiation of contacts manager which attempts to fetch all contacts.
[Relevant Lines](https://github.com/midhunhk/ae-apps-library/blob/master/app/src/main/java/com/ae/apps/common/managers/ContactManager.java#L72-L76)

A ideal fix will be refactoring the lib to be clean i.e when creating a new object refrain from doing other tasks and expose public methods to perform fetching and other similar tasks.